docs/references/config/config.toml.md

```diff
@@ -0,0 +1,2055 @@
+to limit broadcasts to persistent peer nodes.
+
+For non-persistent peers, if enabled, a value of 10 is recommended based on experimental performance results using the
+default P2P configuration.
+
+## State synchronization
+State sync rapidly bootstraps a new node by discovering, fetching, and restoring a state machine snapshot from peers
+instead of fetching and replaying historical blocks. It requires some peers in the network to take and serve state
+machine snapshots. State sync is not attempted if the node has any local state (LastBlockHeight > 0).
+
+The node will have a truncated block history, starting from the height of the snapshot.
+
+### statesync.enable
+Enable state synchronization.
+```toml
+enable = false
+```
+
+| Value type          | boolean |
+|:--------------------|:--------|
+| **Possible values** | `false` |
+|                     | `true`  |
+
+Enable state synchronization on first start.
+
+### statesync.rpc_servers
+Comma-separated list of RPC servers for light client verification of the synced state machine,
+and retrieval of state data for node bootstrapping.
+```toml
+rpc_servers = ""
+```
+
+| Value type                        | string (comma-separated list)      |
+|:----------------------------------|:-----------------------------------|
+| **Possible values within commas** | nodeID@IP:port (`"1.2.3.4:26657"`) |
+|                                   | `""`                               |
+
+At least two RPC servers have to be defined for state synchronization to work.
+
+### statesync.trust_height
+The height of the trusted header hash.
+```toml
+trust_height = 0
+```
+
+| Value type          | integer |
+|:--------------------|:--------|
+| **Possible values** | &gt;= 0 |
+
+`0` is only allowed when state synchronization is disabled.
+
+### statesync.trust_hash
+Header hash obtained from a trusted source.
+```toml
+trust_hash = ""
+```
+
+| Value type          | string             |
+|:--------------------|:-------------------|
+| **Possible values** | hex-encoded number |
+|                     | ""                 |
+
+`""` is only allowed when state synchronization is disabled.
+
+This is the header hash value obtained from the trusted source at height
+[statesync.trust_height](#statesynctrust_height).
+
+### statesync.trust_period
+The period during which validators can be trusted.
+```toml
+trust_period = "168h0m0s"
+```
+
+| Value type          | string (duration) |
+|:--------------------|:------------------|
+| **Possible values** | &gt;= `"0s"`      |
+
+For Cosmos SDK-based chains, `statesync.trust_period` should usually be about 2/3rd of the unbonding period
+(about 2 weeks) during which they can be financially punished (slashed) for misbehavior.
+
+### statesync.discovery_time
+Time to spend discovering snapshots before initiating a restore.
+```toml
+discovery_time = "15s"
+```
+
+<!--- What happens when this time expires? --->
```

Comment:
```suggestion
If `discovery_time` is &gt; 0 and  &lt; 5 seconds, its value will be overridden to 5 seconds. 

If `discovery_time` is zero, the node will not wait for replies once it has broadcast the "snapshot request" message to its peers. If no snapshot data is received, state sync will fail without retrying.

If `discovery_time` is &gt;= 5 seconds, the node will broadcast the "snapshot request" message to its peers and then wait for `discovery_time`. If no snapshot data has been received after that period, the node will retry: it will broadcast the "snapshot request" message again and wait for `discovery_time`, and so on.
```